pkg/kubelet/dockershim/libdocker/kube_docker_client.go

```diff
@@ -582,23 +582,39 @@ func (d *kubeDockerClient) holdHijackedConnection(tty bool, inputStream io.Reade
 	receiveStdout := make(chan error)
 	if outputStream != nil || errorStream != nil {
 		go func() {
+			defer close(receiveStdout)
 			receiveStdout <- d.redirectResponseToOutputStream(tty, outputStream, errorStream, resp.Reader)
 		}()
 	}
 
-	stdinDone := make(chan struct{})
+	stdinDone := make(chan error)
 	go func() {
+		defer close(stdinDone)
 		if inputStream != nil {
-			io.Copy(resp.Conn, inputStream)
+			_, err := io.Copy(resp.Conn, inputStream)
+			if err != nil {
+				stdinDone <- err
+				return
+			}
+		}
+		err := resp.CloseWrite()
```

Comment:
Should this be in a defer block too? It looks like it could be skipped with the return above.